controller/src/main/java/io/pravega/controller/server/ControllerService.java

```diff
@@ -384,9 +387,24 @@ public CompletableFuture<CreateStreamStatus> createStream(String scope, String s
             return CompletableFuture.completedFuture(
                     CreateStreamStatus.newBuilder().setStatus(CreateStreamStatus.Status.INVALID_STREAM_NAME).build());
         }
+        try {
+            validateStreamConfig(streamConfig);
```

Comment:
method implementation - **protected void validateStreamConfig(StreamConfiguration streamConfig) {** This method implementation does not throw any exception but true/false as per condition so why we are catching exception here? is something i'm missing here @shshashwat  @Bhupender-Y ?

Reply:
We have **Preconditions.checkArgument** in **validateStreamConfig**. If it will not fulfill condition which we provided then **validateStreamConfig** will through **IllegalArgumentException**